msc-generator 6.3.9 (new formula)

tmpgnh opened the pull request:
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----
Adding a formula for [Msc-generator](https://sourceforge.net/p/msc-generator) which is a rewrite and expansion of already brewed `mscgen`, with a compatibility mode for the latter.

carlocab commented on Formula/msc-generator.rb:
```diff
@@ -0,0 +1,56 @@
+  depends_on "graphviz"
+
+  # A single change to move out inline definition of a templated member function -- GCC can do it
+  patch :p1, :DATA
```
New formulae need to be able to build without patches. Has this been submitted upstream?

tmpgnh replied:
OK. I will wait for that.

carlocab commented on Formula/msc-generator.rb:
```diff
@@ -0,0 +1,56 @@
+  patch :p1, :DATA
+
+  def install
+    ENV["PATH"] = "#{ENV["HOMEBREW_OPT"]}/gcc/bin:#{ENV["PATH"]}"
```
Superenv takes care of managing the build environment `PATH`. Does this not build with system Clang? Note that you've not added `gcc` as a build dependency, so this has no effect on the build.

tmpgnh replied:
Yes, this is leftover. Will remove.

carlocab commented on Formula/msc-generator.rb:
```diff
@@ -0,0 +1,56 @@
+  def install
+    ENV["PATH"] = "#{ENV["HOMEBREW_OPT"]}/gcc/bin:#{ENV["PATH"]}"
+    system "./configure",
+             "CPPFLAGS=-I#{ENV["HOMEBREW_OPT"]}/glpk/include",
```
You haven't added `glpk` as a build dependency. I don't think this would be needed if you did.

carlocab commented on Formula/msc-generator.rb:
```diff
@@ -0,0 +1,56 @@
+  end
+
+  test do
+    system "#{bin}/msc-gen", "--version"
```
Can you propose a more substantial test than checking the version or calling for help? These tests rarely ever fail even for broken installations.

>We want tests that don’t require any user input and test the basic functionality of the application. For example `foo build-foo input.foo` is a good test and (despite their widespread use) `foo --version` and `foo --help` are bad tests. 

Ref. https://docs.brew.sh/Formula-Cookbook#add-a-test-to-the-formula

tmpgnh replied:
Will add a real one.

carlocab reviewed (commented):
Thanks for your PR, @tmpgnh. A few comments.

carlocab commented on Formula/msc-generator.rb:
```diff
@@ -0,0 +1,56 @@
+  depends_on "help2man" => :build
+  depends_on "pkg-config" => :build
+  depends_on "cairo"
+  depends_on "graphviz"
```
```suggestion
  depends_on "glpk"
  depends_on "graphviz"
```
And you can remove the `CPPFLAGS` argument.

tmpgnh replied:
Done

SMillerDev commented on Formula/msc-generator.rb:
```diff
@@ -0,0 +1,30 @@
+
+  test do
+    system "#{bin}/msc-gen", "--version"
+    system "echo 'a->b;' | #{bin}/msc-gen -S signalling | grep PNG"
```
```suggestion
    assert_match "PNG", pipe_output("#{bin}/msc-gen -S signalling", "a->b;")
```

SMillerDev replied:
this way is more in line with Homebrews usual tests.

tmpgnh commented:
Hi,
I believe I addressed all comments but tests still will not pass. If I get it right, the bison3 brew dependency is the problem, but upstream definitely requires Bison3+. I have checked it and does not compile with the version 2.x which is in MacOS.
Any suggestions?
Thanks

carlocab reviewed (approved):
I think this is fine now. Thanks, @tmpgnh!
